modules/build/src/main/scala/scala/build/options/BuildOptions.scala

```diff
@@ -334,21 +342,85 @@ final case class BuildOptions(
             Right(validVersions.max.repr)
       }
     }
-    val maybeSv = scalaVersion match {
+    val maybeSv = scalaVersionArg match {
       case None => matchNewestScalaVersion(None)
       case Some(sv0) =>
         if (Util.isFullScalaVersion(sv0)) Right(sv0)
         else matchNewestScalaVersion(Some(sv0))
     }
 
-    val sv  = value(maybeSv)
-    val sbv = scalaBinaryVersion.getOrElse(ScalaVersion.binary(sv))
-    (sv, sbv)
+    val scalaVersion       = value(maybeSv)
+    val scalaBinaryVersion = scalaBinaryVersionArg.getOrElse(ScalaVersion.binary(scalaVersion))
+    (scalaVersion, scalaBinaryVersion)
   }
 
+  /** @return
+    *   Either a BuildException or the calculated (ScalaVersion, ScalaBinaryVersion) tuple
+    */
+  private def computeLatestScalaThreeNightlyVersions(): Either[BuildException, (String, String)] =
+    either {
+      import coursier.Versions
+      import coursier.core.Latest
+      import coursier._
+      import scala.concurrent.ExecutionContext.{global => ec}
+      val moduleVersion: Option[String] = {
+        def scala3 = mod"org.scala-lang:scala3-library_3"
+        val res = finalCache.logger.use {
+          Versions()
+            .withModule(scala3)
+            .result()
+            .unsafeRun()(ec)
+        }
+        res.versions.latest(Latest.Release)
+      }
+      val scalaVersion       = moduleVersion.get
```

Comment:
We shouldn't call `Option.get` here. Better to something like
```suggestion
      val scalaVersion       = value(moduleVersion.toRight("No latest version found for $scala3"))
```
(and make `scala3`, which should be a `val`, above accessible from here)

Reply:
Actually, we might need to create a new `BuildException` sub-class in `scala.build.errors`, and create it here
```suggestion
      val scalaVersion       = value(moduleVersion.toRight(new NoScalaVersionFoundError(scala3)))
```